### DISCUSSION TITLE
Get charging cost

### DISCUSSION THREAD

**Discussion by fluxxsimon** [BeatsuDev/tibber.py#42] [Q&A] [answered]
Hi!

Does this API support getting the cost for car charging? I would like to use this API to write a script that calculates my average cost for charging my car.

Br,
Simon

**Comment by ericbstie** [comment, marked as answer]
You can get the price per kWh using tibber.py, but you won't be able to get any information about your car charging status or how much it has charged over the past X hours using tibber.py (or the Tibber API) afaik. If you know the power (Wattage) of your car charger, and it's constant (ish) during charging, you can get the cost of charging by multiplying the current electricity price (in currency/kWh) with the wattage of the car charger (in kW) multiplied with the amount of time charged (in hours). 

The current price can be found like this:

```py
import tibber

account = tibber.Account("<tibber token>")

home = account.homes[0]  # Choose a home
subscription = home.subscriptions[0]  # Choose a subscription
price = subscription.price_info.current
```
The price variable is of [type `Price`](https://github.com/BeatsuDev/tibber.py/blob/fbd4dcea54c578d644a0b9ecc94ed9d72d77997f/tibber/types/price.py#L9) (click the link to see all the properties defined in Price), so you can get the total price, tax, energy price without tax etc. like this:
```py
print(price.total)
print(price.tax)
print(price.energy)
print(price.currency)
print(price.starts_at)
```

To get the hourly prices for electricity today and tomorrow, you can do this:
```
prices_today = subscription.price_info.today
prices_tomorrow = subscription.price_info.tomorrow
```
It will return lists of Price objects. It might come in handy to know that the `price.starts_at` property is a string and to retrieve the datetime object from it, you can use the `datetime.datetime.fromisoformat()`.

Good luck!

**Reply by fluxxsimon** [reply]
Thanks for the answer!

Okay, I was looking to get the historical values. Even if I opt for your solution that you kindly provided, then there is no way of knowing how much that is charged since I run Smart Charging, it kind of lives it own life. 

Do you know if there is a way to get the invoices for my account? There the amount of smart charing I have used is provided, and I could maybe parse some text file, if that's available.

Best regards,
Simon

**Reply by ericbstie** [reply]
@fluxxsimon Is your smart charging expense shown on your invoice from Tibber? If so, is your invoice available in the Tibber app or elsewhere? If elsewhere, maybe you can read the data from there? The Tibber API does not expose invoices through their API afaik so you won't be able to get that data using tibber.py as of now. Reading information outside of the Tibber API is out of the scope of tibber.py, so I won't be able to help you unfortunately. Maybe Tibber support can help you further or forward a request to add invoices to the Tibber API (and eventually be integrated to tibber.py)!

As for reading historical price data, you can do that with tibber.py like this:

```python
consumption = home.fetch_consumption(resolution = "HOURLY", last = 24)  # Fetches last 24 hours
consumption.nodes # List of Consumption objects fetched

consumption.nodes[-1].unit_price  # The price per kWh for the last hour
```

Here are the attributes of Consumption: https://github.com/BeatsuDev/tibber.py/blob/master/tibber/types/consumption.py#L11
